lib/libvfio-user.c

```diff
@@ -698,26 +699,28 @@ MOCK_DEFINE(get_next_command)(vfu_ctx_t *vfu_ctx, struct vfio_user_header *hdr,
     /* FIXME get request shouldn't set errno, it should return it as -errno */
     ret = vfu_ctx->tran->get_request(vfu_ctx, hdr, fds, nr_fds);
     if (unlikely(ret < 0)) {
-        if (ret == -EAGAIN || ret == -EWOULDBLOCK) {
+        switch (-ret) {
+        case EAGAIN:
             return 0;
-        }
-        if (ret != -EINTR) {
+
+        case ENOMSG:
+            vfu_reset_ctx(vfu_ctx, "closed");
+            return -ENOTCONN;
+
+        case ECONNRESET:
+            vfu_reset_ctx(vfu_ctx, "reset");
+            return -ENOTCONN;
+
+        case EINTR:
+            return -EINTR;
```

Comment:
Should we have a log message here? Useful for debugging.

Reply:
yeah, good idea.